pennylane/transforms/hamiltonian_expand.py

```diff
@@ -129,20 +129,26 @@ def hamiltonian_expand(tape, group=True):
             "Passed tape must end in `qml.expval(H)`, where H is of type `qml.Hamiltonian`"
         )
 
+    # note: for backward passes of some frameworks
+    # it is crucial to use the hamiltonian.data attribute,
+    # and not hamiltonian.coeffs when recombining the results
```

Comment:
I started to leave notes like this, because that piece of information could take a future developer a week to figure out! I feel even if there are a few redundant notes at some stage, it is better than withholding this info!